core/finality-grandpa/src/lib.rs

```diff
@@ -683,12 +684,16 @@ pub fn run_grandpa_voter<B, E, Block: BlockT<Hash=H256>, N, RA, SC, X>(
 					aux_schema::write_voter_set_state(&**client.backend(), &set_state)?;
 
 					let set_state: SharedVoterSetState<_> = set_state.into();
-
+					let voters: VoterSet<_> = new.authorities.into_iter().collect();
+					let local_key = session_store
+						.get_keys(|public| env.voters.contains_key(public))
```

Comment:
should rather be `new.authorities.contains_key`, right? this is setting the key based off of the _previous_ authority set where it should be based on the new authority set.